magenta-lib/src/main/scala/magenta/input/DeploymentResolver.scala

```diff
@@ -0,0 +1,68 @@
+package magenta.input
+
+import play.api.libs.json.JsValue
+
+case class Deployment(
+  name: String,
+  `type`: String,
+  stacks: List[String],
+  regions: List[String],
+  dependencies: List[String],
+  app: String,
+  contentDirectory: String,
+  parameters: Map[String, JsValue]
+)
+
+object DeploymentResolver {
+  val DEFAULT_REGIONS = List("eu-west-1")
+
+  def resolveDeployment(name: String, deployment: DeploymentOrTemplate, yaml: RiffRaffYaml): Either[List[String], DeploymentOrTemplate] = {
+    val missingDeps = yaml.missingDependencies(deployment.dependencies.getOrElse(Nil))
+    if (missingDeps.nonEmpty)
+      Left(missingDeps.map(dep => s"Missing dependency $dep in $name"))
```

Comment:
No - the template with name error catches when the `template` field is referencing a non-existent template in the `templates` section. This is catching when any of the `dependencies` lists are referring to a deployment that doesn't exist. They are similar but not the same.